eng/templates/official/jobs/build-artifacts.yml

```diff
@@ -1,80 +1,212 @@
 jobs:
-  - job: "Build"
-    displayName: "Build python worker"
+- job: Build_WINDOWS_X64
+  strategy:
+    matrix:
+      Python37V4:
+        pythonVersion: '3.7'
+        workerPath: 'python/prodV4/worker.py'
+      Python38V4:
+        pythonVersion: '3.8'
+        workerPath: 'python/prodV4/worker.py'
+      Python39V4:
+        pythonVersion: '3.9'
+        workerPath: 'python/prodV4/worker.py'
+      Python310V4:
+        pythonVersion: '3.10'
+        workerPath: 'python/prodV4/worker.py'
+      Python311V4:
+        pythonVersion: '3.11'
+        workerPath: 'python/prodV4/worker.py'
+  templateContext:
+    outputParentDirectory: $(Build.ArtifactStagingDirectory)
+    outputs:
+      - output: pipelineArtifact
+        targetPath: $(Build.SourcesDirectory)
+        artifactName: "$(pythonVersion)_WINDOWS_X64"
+  steps:
+  - template: ../../../../pack/templates/win_env_gen.yml
+    parameters:
+      pythonVersion: '$(pythonVersion)'
+      workerPath: '$(workerPath)'
+      architecture: 'x64'
+      artifactName: '$(pythonVersion)_WINDOWS_X64'
+- job: Build_WINDOWS_X86
+  strategy:
+    matrix:
+      Python37V4:
+        pythonVersion: '3.7'
+        workerPath: 'python/prodV4/worker.py'
+      Python38V4:
+        pythonVersion: '3.8'
+        workerPath: 'python/prodV4/worker.py'
+      Python39V4:
+        pythonVersion: '3.9'
+        workerPath: 'python/prodV4/worker.py'
+      Python310V4:
+        pythonVersion: '3.10'
+        workerPath: 'python/prodV4/worker.py'
+      Python311V4:
+        pythonVersion: '3.11'
+        workerPath: 'python/prodV4/worker.py'
+  templateContext:
+    outputParentDirectory: $(Build.ArtifactStagingDirectory)
+    outputs:
+      - output: pipelineArtifact
+        targetPath: $(Build.SourcesDirectory)
+        artifactName: "$(pythonVersion)_WINDOWS_X86"
+  steps:
+  - template: ../../../../pack/templates/win_env_gen.yml
+    parameters:
+      pythonVersion: '$(pythonVersion)'
+      workerPath: '$(workerPath)'
+      architecture: 'x86'
+      artifactName: '$(pythonVersion)_WINDOWS_x86'
+- job: Build_LINUX_X64
```

Comment:
Is there no pool for each job similar to github? Here linux and OSX_64 are running the same template nix_env_gen.yml, so I guess they should be running on different OS's?

Reply:
Updated